common/lib/xmodule/xmodule/x_module.py

```diff
@@ -766,7 +768,13 @@ def public_view(self, _context):
             u'<span class="icon icon-alert fa fa fa-warning" aria-hidden="true"></span>'
             u'<div class="message-content">{}</div></div></div>'
         )
-        return Fragment(alert_html.format(DEFAULT_PUBLIC_VIEW_MESSAGE))
+
+        if self.display_name:
+            display_text = DEFAULT_PUBLIC_VIEW_MESSAGE %self.display_name
+        else:
+            display_text = DEFAULT_PUBLIC_VIEW_MESSAGE %"This content"
```

Comment:
This is not going to work with the translation system so you will need a string each for the two cases.